dali/python/nvidia/dali/external_source.py

```diff
@@ -155,81 +155,98 @@ class ExternalSource():
+    explicitly supply the stream handle.
+
+    Here are the special values:
+      *  0 - Use the default CUDA stream
+      * -1 - Use DALI's internal stream
+
+    Since you cannot synchronize with this stream to prevent overwriting the array with new data
+    in another stream, if an internal stream is used, the call to ``feed_input`` will not complete
+    until the copy to the internal buffer is complete.
 
 `use_copy_kernel` : optional, `bool`
-    If set to True, DALI will use a CUDA kernel to feed the data (only applicable when copying
-    data to/from GPU memory) instead of cudaMemcpyAsync (default).
+    If set to True, DALI will use a CUDA kernel to feed the data instead of cudaMemcpyAsync (default).
+
+    .. note::
+        This is applicable only when copying data to and from GPU memory.
+
+`blocking` : optional,
+   Determines whether the external source should not complete until data is available or just fail
+   when the is not available.
 
-`blocking` : optional, Whether external source should block until data is available or just
-    fail when it is not
+`no_copy` : Determines whether DALI should copy the buffer when feed_input is called.
 
-`no_copy` : Whether DALI should copy the buffer when feed_input is called
-    If True, DALI passes the user memory directly to the Pipeline, instead of copying.
-    It is the user's responsibility to keep the buffer alive and unmodified
-    until it is consumed by the pipeline.
+    If set to True, DALI passes the user memory directly to the pipeline, instead of copying the
+    memory. It is your responsibility to keep the buffer alive and unmodified until it is
+    consumed by the pipeline.
 
-    The buffer can be modified or freed again after the relevant iteration output has been consumed.
-    (when they are not equal) iterations following the``feed_input`` call.
-    Effectively, it happens after ``prefetch_queue_depth`` or ``cpu_queue_depth * gpu_queue_depth``
+    The buffer can be modified or freed again after the relevant number of iteration output
+    has been consumed. Effectively, it happens after ``prefetch_queue_depth`` or
+    ``cpu_queue_depth * gpu_queue_depth`` (when they are not equal) iterations following
+    the``feed_input`` call.
 
-    Provided memory must match the specified `device` parameter of the operator.
-    For CPU, the provided memory can be one contiguous buffer or a list of contiguous Tensors.
-    For GPU to not do any copies the provided buffer must be contiguous. If user provides a list
-    of separate Tensors there will be an additional internal copy made.
+    The memory that is provided memory must match the specified ``device`` parameter of the operator.
+    For the CPU, the provided memory can be one contiguous buffer or a list of contiguous Tensors.
+    For the GPU, to prevent copying, the provided buffer must be contiguous. If you provides a list
+    of separate Tensors, there will be an additional internal copy is created.
```

Comment:
```suggestion
  For the GPU, to avoid extra copy, the provided buffer must be contiguous. If you provides a list
  of separate Tensors, there will be an additional internal copy made, consuming both memory
  and bandwidth.)code", false)
```

Reply:
Done